### PR TITLE
Feat/geo2024

### PR DESCRIPTION
**Pull request by Datayama38**
## Summary by CodeRabbit

- **New Features**
	- Enhanced migration commands with additional flags for improved functionality.
	- Introduced new datasets for 2024, including various `Insee`, `Dgcl`, and `Eurostat` datasets.
	- Added integration tests for the new datasets ensuring their functionality.

- **Bug Fixes**
	- Improved resource management by ensuring database connections are properly closed in test suites.

- **Documentation**
	- Updated test specifications to reflect new dataset functionalities and validation processes.

- **Refactor**
	- Streamlined the `migrate` function and improved logging for better traceability.
	- Adjusted import statements for clarity and organization.

- **Tests**
	- Added and modified integration tests for multiple datasets to ensure comprehensive coverage.

### CODE REVIEW

File: api/src/etl/Migrator.ts [hunk 1]
```diff
@@ -139,7 +139,9 @@ export class Migrator {
       );
     } catch (e) {
```
Comment (jonathanfallon) [inline api/src/etl/Migrator.ts:140]: Tu ne peux pas typer l'erreur ici directement ?

File: api/src/etl/buildMigrator.integration.spec.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ import { CreateComEvolutionTable } from "./datastructure/001_CreateComEvolutionT
 import { createPool } from "./helpers/index.ts";
 import { MemoryStateManager } from "./providers/MemoryStateManager.ts";
 
-describe.skip("Build Migrator", () => {
+Deno.test("Build Migrator", () => {
```
Comment (jonathanfallon) [inline api/src/etl/buildMigrator.integration.spec.ts:44]: Le `describe` ne fonctionne pas ?

Comment (Datayama38) [reply api/src/etl/buildMigrator.integration.spec.ts:44]: Si, j'étais fatigué vendredi... j'avais pas vu qu'ils étaient skip... c'est pour ça qu'ils ne marchaient pas
Je corrige

File: api/deno.lock [hunk 3]
```diff
@@ -1,4068 +1,4050 @@
...
-      },
-      "zip-stream@4.1.1": {
-        "integrity": "sha512-9qv4rlDiopXg4E69k+vMHjNN63YFMe9sZMrdlvKnCjlCRWeCBswPPMPUfx+ipsAWq1LXHe70RcbaHdJJpS6hyQ==",
-        "dependencies": {
-          "archiver-utils": "archiver-utils@3.0.4",
-          "compress-commons": "compress-commons@4.1.2",
-          "readable-stream": "readable-stream@3.6.2"
-        }
-      }
+  "version": "4",
```
Comment (P3rceval) [inline api/deno.lock:2]: @Datayama38 Faut que tu repasses en version 3 je crois

**Review by P3rceval — APPROVED** [review]

**Pull request merged**